pkg/ccl/kvccl/kvfollowerreadsccl/followerreads_test.go

```diff
@@ -674,19 +687,34 @@ func TestOracle(t *testing.T) {
 			st := cluster.MakeTestingClusterSettings()
 			kvserver.FollowerReadsEnabled.Override(ctx, &st.SV, !c.disabledFollowerReads)
 
-			o := replicaoracle.NewOracle(followerReadOraclePolicy, replicaoracle.Config{
-				NodeDescs:  nodes,
-				Settings:   st,
-				RPCContext: rpcContext,
-				Clock:      clock,
-				HealthFunc: func(roachpb.NodeID) bool { return true },
-			})
+			o := replicaoracle.NewOracle(followerReadOraclePolicy, cfg(st))
 
 			res, _, err := o.ChoosePreferredReplica(ctx, c.txn, desc, c.lh, c.ctPolicy, replicaoracle.QueryState{})
 			require.NoError(t, err)
 			require.Equal(t, c.exp, res)
 		})
 	}
+
+	t.Run("bulk", func(t *testing.T) {
+		st := cluster.MakeTestingClusterSettings()
+		ctx := context.Background()
+		var noTxn *kv.Txn
+		var noLeaseholder *roachpb.ReplicaDescriptor
+		var noCTPolicy roachpb.RangeClosedTimestampPolicy
+		var noQueryState replicaoracle.QueryState
+
+		t.Run("no-filter", func(t *testing.T) {
+			br := NewBulkOracle(cfg(st), roachpb.Locality{})
+			_, _, err := br.ChoosePreferredReplica(ctx, noTxn, desc, noLeaseholder, noCTPolicy, noQueryState)
+			require.NoError(t, err)
+		})
+		t.Run("filter", func(t *testing.T) {
+			br := NewBulkOracle(cfg(st), region("b"))
+			picked, _, err := br.ChoosePreferredReplica(ctx, noTxn, desc, noLeaseholder, noCTPolicy, noQueryState)
```

Comment:
let's add a test that tries to choose prefer replica when no locality match exists.